src/vm.rs

```diff
@@ -1621,55 +1637,47 @@ impl<'a, T: GuestStagePagingMode> FinalizedVm<'a, T> {
         Ok(0)
     }
 
-    #[allow(clippy::too_many_arguments)]
     fn guest_get_evidence(
         &self,
-        cert_request_addr: u64,
-        cert_request_size: usize,
+        csr_guest_addr: u64,
+        csr_len: usize,
         _request_data_addr: u64,
         _evidence_format: u64,
-        cert_addr_out: u64,
-        cert_size: usize,
-        active_pages: &ActiveVmPages<T>,
+        certout_guest_addr: u64,
+        certout_len: usize,
     ) -> EcallResult<u64> {
-        if cert_request_size > MAX_CSR_LEN {
-            return Err(EcallError::Sbi(SbiError::InsufficientBufferCapacity));
+        // Map CSR read-only.
+        let (csr_addr, _csr_mapping) = self.map_guest_range_in_umode_slot(
+            UmodeSlotId::A,
+            csr_guest_addr,
+            csr_len as u64,
+            false,
```

Comment:
pre-existing, and doesn't have to be part of this PR: I assume the `bool` here means RO vs RW? Can we make that an enum instead? :)

Reply:
Ah that's a great suggestion. Will change.

Reply:
Done.